### PR TITLE
Update app dependencies

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/608920331025315/1169361282385305/f
Tech Design URL: 
CC: 

**Description**:
Updated most of the libraries which had newer versions available. A full list (along with some of the highlights of the new versions available in the linked Asana task)

**Steps to test this PR**:
1. This requires **thorough** manual testing of the whole app. Collectively, these libraries have a large footprint of potential change and therefore we need to ensure the whole app is behaving (and looking) as we'd expect it to.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -95,41 +95,39 @@ android {
...
-    androidTestRunner = "1.0.2"
-    testRunner = "1.1.1"
-    coroutines = "1.3.1"
+    coreTesting = "2.1.0"
+    testRunner = "1.2.0"
+    coroutines = "1.3.5"
     retrofitCoroutinesAdapter = "0.9.2"
-    webkit = "1.1.0"
+    webkit = "1.2.0"
+    referrerLibrary = "1.1.2"
```
Comment (cmonfortep) [inline app/build.gradle:114]: How can we test this?

File: app/src/androidTest/java/com/duckduckgo/app/httpsupgrade/api/HttpsBloomFilterSpecJsonTest.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ class HttpsBloomFilterSpecJsonTest {
     fun whenGivenValidJsonThenParsesCorrectly() {
         val moshi = Moshi.Builder().build()
         val jsonAdapter = moshi.adapter(HttpsBloomFilterSpec::class.java)
-        val result = jsonAdapter.fromJson(json())
+        val result = jsonAdapter.fromJson(json())!!
```
Comment (cmonfortep) [inline app/src/androidTest/java/com/duckduckgo/app/httpsupgrade/api/HttpsBloomFilterSpecJsonTest.kt:30]: Will this crash if the json is malformed?

**Comment by marcosholgado** [comment]
Not sure if would also make sense to do https://app.asana.com/0/414730916066338/1158589410650834 as part of this. I guess there is no rush but definitely a good candidate for next LHF days.

File: app/build.gradle (hunk 1 above)
Comment (CDRussell) [reply app/build.gradle:114]: i'd recommend building the app and uploading to the Play Store ad hoc distribution page.

- delete the build.gradle line where we add the `debug` suffix
- upload your apk, and get a link
- email this link to yourself, but append on query params to set the referrer
- install the app, watch logcat to ensure the referrer is picked up

(I have manually done the above. If you want to do it too, I can send you the referrer params to add)

File: app/src/androidTest/java/com/duckduckgo/app/httpsupgrade/api/HttpsBloomFilterSpecJsonTest.kt (hunk 2 above)
Comment (CDRussell) [reply app/src/androidTest/java/com/duckduckgo/app/httpsupgrade/api/HttpsBloomFilterSpecJsonTest.kt:30]: Yes, and I believe this is the same behavior as we currently have. In this instance, the test will crash which is no big deal. In production, I think RxJava will catch the problem and it'll be a "sync failed" scenario.

**Comment by malmstein** [comment]
This has been approved for 2 days, any reason why it's not yet merged @cmonfortep?

**Comment by cmonfortep** [comment]
@malmstein we will keep this on hold till we do the next release (coming soon). Check the meeting notes here https://app.asana.com/0/12988952642567/1169184762840727/f where it's explained with more info "Android dependencies release".

**Pull request merged**